building/setup-apt-branch/setup-apt-branch.sh

```diff
@@ -0,0 +1,39 @@
+if [ -z "${HOMEWORLD_APT_BRANCH:-}" ]
+then
+	echo 'Error: Need to specify apt branch:' >&2
+	echo '$ export HOMEWORLD_APT_BRANCH=<username>/<branch>' >&2
+	exit 1
+fi
+
+apt_branch_regex='^[0-9a-zA-Z_-]+\/[0-9a-zA-Z_-]+$'
+if ! [[ "${HOMEWORLD_APT_BRANCH}" =~ $apt_branch_regex ]]
+then
+	echo 'Error: Apt branch invalid. Should be of the form <username>/<branch>.' >&2
+	echo 'Allowed characters: [0-9a-zA-Z_-]' >&2
+	exit 1
+fi
+
+function get_apt_signing_key()
+{
+    while read branch key; do
+        if [ "${branch}" == "${HOMEWORLD_APT_BRANCH}" ] || [ "${branch}" == "$(echo ${HOMEWORLD_APT_BRANCH} | cut -d '/' -f 1)" ]; then
+            HOMEWORLD_APT_SIGNING_KEY="${key}"
+            break
+        fi
+    done < `dirname $BASH_SOURCE`/signing-keys
+
+    if [ -z "${HOMEWORLD_APT_SIGNING_KEY:-}" ]
+    then
+        echo 'error: apt branch not found in signing-keys' >&2
+        exit 1
+    fi
+
+    apt_signing_key_regex='^[0-9a-zA-Z]+$'
+    if ! [[ "${HOMEWORLD_APT_SIGNING_KEY}" =~ $apt_signing_key_regex ]]
+    then
+        echo 'error: apt signing key invalid' >&2
+        exit 1
```

Comment:
Have you confirmed that this error checking works?

Reply:
Yes, I have tested this manually with different inputs.